code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -606,6 +611,99 @@ GLOBAL_VAR(medibot_unique_id_gen)
 /obj/machinery/bot_core/medbot
 	req_one_access = list(ACCESS_MEDICAL, ACCESS_ROBOTICS)
 
+
+/mob/living/simple_animal/bot/medbot/cola
+	name = "Sponsored Medibot"
+	desc = "Even medical isn't safe from corporate greed. May contain cola."
+	icon_state = "colabot0"
+	radio_key = /obj/item/encryptionkey/medical_sponsor  // grants cargo
+	COOLDOWN_STATIC_DECLARE(fund)
+	var/static/budget = 0  // initial value (will get refilled instantly)
+	var/static/refill = 200
+	var/payout = 10
+
+/mob/living/simple_animal/bot/medbot/cola/New()
+	..()
+	START_PROCESSING(SSobj, src)
+
+/mob/living/simple_animal/bot/medbot/cola/process()
+	..()
+	if(COOLDOWN_FINISHED(src, fund))
+		refill()
+		COOLDOWN_START(src, fund, 2 MINUTES)
```

Comment:
I don't think you should call the parent here.
![image](https://user-images.githubusercontent.com/40559528/183282996-0d0d0578-c0f1-4743-9864-ca2a73d98d63.png)

Reply:
oh oops